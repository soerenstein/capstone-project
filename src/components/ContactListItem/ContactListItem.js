import React from 'react'
import styled from 'styled-components/macro'
import { ReactComponent as UserIcon } from '../Design/img/user.svg'

export default function ContactListItem({
  firstName,
  lastName,
  company,
  id,
  onClick,
}) {
  return (
    <ContactContainer key={id} onClick={onClick}>
      <SVGWrapper>
        <UserIcon />
      </SVGWrapper>
      <StyledContactInformation>
        {firstName + ' ' + lastName}
      </StyledContactInformation>
      <StyledContactInformation>{company}</StyledContactInformation>
    </ContactContainer>
  )
}

const ContactContainer = styled.div`
  padding: 10px 30px;
  height: 100%;
  margin: 10px;
  display: grid;
  grid-template-columns: 30% auto;
  justify-self: end;
  border-radius: 10px;
  box-shadow: 8px 6px 20px -10px rgba(128, 128, 128, 1);
`
const SVGWrapper = styled.div`
  grid-column: 1;
  grid-row: 1 / 3;
  justify-self: center;
  align-self: center;
`
const StyledContactInformation = styled.p`
  padding: 5px;
  margin: 0;
  justify-self: start;
  grid-column: 2;
  word-wrap: normal;
  width: 90%;
  overflow: hidden;
  text-overflow: ellipsis;
  height: 1em;
`
