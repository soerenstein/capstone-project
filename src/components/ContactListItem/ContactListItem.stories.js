import React from 'react'
import ContactListItem from './ContactListItem'

export default {
  title: 'ContactListItem',
  component: ContactListItem,
}

const List = (args) => <ContactListItem {...args} />

export const ListItem = List.bind({})
ListItem.args = {
  firstName: 'Max',
  lastName: 'Mustermann',
  company: 'Universal GmbH',
}
