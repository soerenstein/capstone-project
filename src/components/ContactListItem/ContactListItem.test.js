import React from 'react'
import { render } from '@testing-library/react'
import ContactListItem from './ContactListItem'

describe('The listItem', () => {
  it('renders correctly', () => {
    const listItem = render(<ContactListItem />)
    expect(listItem).toMatchSnapshot()
  })
})
