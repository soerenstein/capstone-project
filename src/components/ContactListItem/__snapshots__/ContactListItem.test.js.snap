// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The listItem renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li>
        <section
          class="ContactListItem__ContactContainer-q0ui6m-0 cZncpw"
        >
          <section
            class="ContactListItem__SVGWrapper-q0ui6m-1 CAeIO"
          >
            <svg>
              user.svg
            </svg>
          </section>
          <p
            class="ContactListItem__StyledContactInformation-q0ui6m-2 LyweK"
          >
            undefined undefined
          </p>
          <p
            class="ContactListItem__StyledContactInformation-q0ui6m-2 LyweK"
          />
        </section>
      </li>
    </div>
  </body>,
  "container": <div>
    <li>
      <section
        class="ContactListItem__ContactContainer-q0ui6m-0 cZncpw"
      >
        <section
          class="ContactListItem__SVGWrapper-q0ui6m-1 CAeIO"
        >
          <svg>
            user.svg
          </svg>
        </section>
        <p
          class="ContactListItem__StyledContactInformation-q0ui6m-2 LyweK"
        >
          undefined undefined
        </p>
        <p
          class="ContactListItem__StyledContactInformation-q0ui6m-2 LyweK"
        />
      </section>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
